lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1870,12 +1893,19 @@ def __init__(self, charm, relation_names, relabel_instance=True):
             relabel_instance: A boolean flag indicating if Prometheus
                 scrape job "instance" labels must refer to Juju Topology.
         """
-        super().__init__(charm, relation_names["prometheus"])
-
         self._charm = charm
-        self._target_relation = relation_names["scrape_target"]
-        self._prometheus_relation = relation_names["prometheus"]
-        self._alert_rules_relation = relation_names["alert_rules"]
+
+        relation_names = relation_names or {}
+
+        self._prometheus_relation = relation_names.get(
+            "prometheus", "downstream-prometheus-scrape"
+        )
+        self._target_relation = relation_names.get("scrape_target", "prometheus-target")
+        self._alert_rules_relation = relation_names.get("alert_rules", "prometheus-rules")
```

Comment:
They're somewhat "undocumented 'recommended defaults'" because the only actual user of this class is `cos-proxy`, where they're very clearly in `metadata.yaml`. 

`relation_names` _was_ mandatory previously, which got scraped out in this PR because it's not used by anybody else, and honestly, it probably won't be. The major reason for keeping it at all (instead of just making them constants or kwargs) is to avoid changing the unit tests for prometheus _too much_, given the "only one charm using it" caveat.